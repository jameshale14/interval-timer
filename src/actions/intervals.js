import database from '../firebase/firebase'

//GET_INTERVALS
export const getIntervals = () => ({
  type: 'GET_INTERVALS',
})

//START_GET_INTERVALS

//CREATE_INTERVAL
export const createInterval = (interval) => ({
  type: 'CREATE_INTERVAL',
  interval
})

//START_CREATE_INTERVAL

//UPDATE_INTERVAL
export const updateInterval = (id, updates) => ({
  type: 'UPDATE_INTERVAL',
  id,
  updates
})

//START_UPDATE_INTERVAL

//DELETE_INTERVAL
export const deleteInterval = (id) => ({
  type: 'DELETE_INTERVAL',
  id
})

//START_DELETE_INTERVAL

export const startDeleteInterval = (id) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid
    return database.ref(`users/${uid}/intervals/${id}`).remove()
      .then(() => {
        dispatch(deleteInterval(id))
      })
  }
}

//SET_INTERVALS
export const setIntervals = (intervals) => ({
  type: 'SET_INTERVALS',
  intervals
})

//START_SET_INTERVALS
export const startSetIntervals = () => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid
    return database.ref(`users/${uid}/intervals`).once('value')
      .then((snapshot) => {
        const intervals = []
        snapshot.forEach((childSnapshot) => {
          intervals.push({
            id: childSnapshot.key,
            ...childSnapshot.val()
          })
        })
        dispatch(setIntervals(intervals))
      })
  }
}