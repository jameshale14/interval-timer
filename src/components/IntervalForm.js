import React from 'react'
import PropTypes from 'prop-types'

export default class IntervalForm extends React.Component {

  constructor(props) {
    super(props)
    this.state = {
      stepError: undefined,
      saveError: undefined,
      name: props.interval ? props.interval.name : '',
      steps: props.interval ? props.interval.steps : [],
      newStep: {
        type: 'Activity',
        name: '',
        duration: ''
      }
    }
  }

  handleAddStep = (e) => {
    e.preventDefault()
    if (this.state.newStep.type == '' || this.state.newStep.name == '' || this.state.newStep.duration == 0) {
      this.setState(() => ({ stepError: 'Please fill in all fields' }))
    } else {
      this.setState((prevState) => {
        const newStep = { ...this.state.newStep }
        return {
          steps: [...prevState.steps, newStep],
          stepError: undefined
        }
      })
      // this.setState(() => ({  }))
    }
  }

  onTypeChange = (e) => {
    const type = e.target.value
    this.setState((prevState) => {
      const newStep = prevState.newStep
      newStep.type = type
      return { newStep }
    })
  }

  onNameChange = (e) => {
    const name = e.target.value
    this.setState((prevState) => {
      const newStep = prevState.newStep
      newStep.name = name
      return { newStep }
    })
  }

  onIntervalNameChange = (e) => {
    const name = e.target.value
    this.setState(() => ({ name }))
  }

  onDurationChange = (e) => {
    const duration = e.target.value
    this.setState((prevState) => {
      const newStep = prevState.newStep
      newStep.duration = duration
      return { newStep }
    })
  }

  handleOnSave = (e) => {
    e.preventDefault()

    if (!this.state.name || this.state.steps.length === 0) {
      this.setState(() => ({ saveError: 'Please provide a routine name and interval steps.' }))
    } else {
      this.setState(() => ({ saveError: undefined }))

      const interval = {
        name: this.state.name,
        steps: this.state.steps
      }

      this.props.onSubmit(interval)
    }

  }

  handleRemoveStep = (e) => {
    e.preventDefault()
    const stepIndex = parseInt(e.currentTarget.value)
    this.setState((prevState) => {
      const steps = prevState.steps.filter((step, index) => {
        return index != stepIndex
      })

      return ({
        steps
      })

    })
  }

  render() {
    return (
      <div>
        <button id='saveInterval' onClick={this.handleOnSave}>Save</button>
        {this.state.saveError && <p>{this.state.saveError}</p>}
        <input
          type='text'
          placeholder='name'
          value={this.state.name}
          onChange={this.onIntervalNameChange}
        />
        {
          this.state.steps.map((step, stepIndex) => {
            return (
              <div key={stepIndex} >
                <p>Type: {step.type}</p>
                <p>Name: {step.name}</p>
                <p>Duration: {step.duration}</p>
                <button onClick={this.handleRemoveStep} value={stepIndex}>Remove</button>
              </div>
            )
          })
        }
        <div>
          {this.state.stepError && <p>{this.state.stepError}</p>}
          <form onSubmit={this.handleAddStep}>
            <select
              value={this.state.newStep.type}
              onChange={this.onTypeChange}
              onBlur={this.onTypeChange}
            >
              <option value='Activity'>Activity</option>
              <option value='Rest'>Rest</option>
            </select>

            <input
              type='text'
              placeholder='Step name'
              value={this.state.newStep.name}
              onChange={this.onNameChange}
            />
            <input
              type='number'
              placeholder='Duration (seconds)'
              step='1'
              value={this.state.newStep.duration}
              onChange={this.onDurationChange}
            />
            <button onClick={this.handleAddStep}>Add Step</button>
          </form>
        </div>
      </div>
    )
  }
}

IntervalForm.propTypes = {
  interval: PropTypes.object,
  onSubmit: PropTypes.func
}
