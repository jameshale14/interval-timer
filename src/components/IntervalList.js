import React from 'react'
import { connect } from 'react-redux'

export const IntervalList = (props) => (
  <div>
    {props.intervals.map((interval) => {
      return (<div>{interval.name}</div>)
    })}
  </div>
)

const mapStateToProps = (state) => ({
  intervals: []
})

export default connect(mapStateToProps)(IntervalList)