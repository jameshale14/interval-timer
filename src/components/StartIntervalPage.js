/*eslint-env browser*/

import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

export class StartIntervalPage extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      currentStepIndex: 0,
      currentStepName: undefined,
      currentTimeRemaining: undefined,
      isRunning: false
    }
    this.interval = undefined
  }

  componentDidMount() {
    document.title = 'Start Interval | Interval Timer'
  }

  componentDidUpdate() {
    document.title = 'Start Interval | Interval Timer'
  }

  initialise = (stepIndex) => {
    // const currentStepIndex = this.state.currentStepIndex
    this.setState(() => ({
      currentStepName: this.props.interval.steps[stepIndex].name,
      currentTimeRemaining: this.props.interval.steps[stepIndex].duration
    }))
  }

  runTimer = () => {
    if (!this.state.isRunning) {
      return
    }

    const currTime = this.state.currentTimeRemaining - 1
    this.playSound(currTime)

    if (this.state.currentTimeRemaining == 1) {
      if (this.state.currentStepIndex < (this.props.interval.steps.length - 1)) {
        this.setState((prevState) => ({
          currentStepIndex: (prevState.currentStepIndex + 1)
        }))
        this.initialise(this.state.currentStepIndex)
      } else {
        window.clearInterval(this.interval)
        this.setState(() => ({ isRunning: false }))
      }

    } else {
      // const currTime = this.state.currentTimeRemaining

      this.setState((prevState) => ({
        currentTimeRemaining: (prevState.currentTimeRemaining - 1)
      }))

      // this.playSound(currTime)

    }
  }

  playSound = (timeRemaining) => {
    if (timeRemaining > 3) {
      return
    }

    if (timeRemaining == 0) {
      const AudioContext = window.AudioContext || window.webkitAudioContext
      const context = new AudioContext()
      const g = context.createGain()
      const o = context.createOscillator()
      o.type = 'sine'
      o.connect(g)
      o.frequency.value = 659.26
      g.connect(context.destination)
      o.start(0)
      g.gain.exponentialRampToValueAtTime(0.00001, context.currentTime + 2.5)

    } else {
      const AudioContext = window.AudioContext || window.webkitAudioContext
      const context = new AudioContext()
      const g = context.createGain()
      const o = context.createOscillator()
      o.type = 'sine'
      o.connect(g)
      o.frequency.value = 440.0
      g.connect(context.destination)
      o.start(0)
      g.gain.exponentialRampToValueAtTime(0.00001, context.currentTime + 1)

    }
  }

  handleStart = (e) => {
    e.preventDefault()
    if (this.state.isRunning) {
      return
    }
    if (this.state.currentTimeRemaining == undefined) {
      this.initialise(0)
    }
    this.setState(({ isRunning: true }))
    this.interval = setInterval(this.runTimer, 1000)

  }

  handleStop = (e) => {
    e.preventDefault()

    window.clearInterval(this.interval)
    this.setState(() => ({ isRunning: false }))
  }

  handleRestartStep = (e) => {
    e.preventDefault()

    this.initialise(this.state.currentStepIndex)
    this.setState(({ isRunning: true }))
    this.interval = setInterval(this.runTimer, 1000)

  }

  handleRestartWholeInterval = (e) => {
    e.preventDefault()

    this.setState(() => ({ currentStepIndex: 0 }))
    this.initialise(0)
    this.setState(({ isRunning: true }))
    this.interval = setInterval(this.runTimer, 1000)

  }

  render() {
    return (
      <div>
        <h3 data-testid='interval-name'>{this.props.interval.name}</h3>
        <h1 data-testid='step-name'>{this.state.currentStepName}</h1>
        <h1 data-testid='time-remaining'>{this.state.currentTimeRemaining}</h1>
        <button onClick={this.handleStart}>Start</button>
        <button onClick={this.handleStop}>Stop</button>
        {!this.state.isRunning && this.state.currentStepName != undefined &&
          <>
            <button onClick={this.handleRestartStep}>Restart Step</button>
            <button onClick={this.handleRestartWholeInterval}>Restart Interval</button>
          </>
        }
      </div>
    )
  }

}

StartIntervalPage.propTypes = {
  interval: PropTypes.object
}

const mapStateToProps = (state, props) => ({
  interval: state.intervals.find((interval) => interval.id === props.match.params.id)
})

export default connect(mapStateToProps)(StartIntervalPage)