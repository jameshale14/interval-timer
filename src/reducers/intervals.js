export default (state = {}, action) => {
  switch (action.type) {
    case 'GET_INTERVALS':
      return state
    case 'CREATE_INTERVAL':
      return [...state, action.interval]
    case 'UPDATE_INTERVAL':
      return state.map((interval) => {
        if (interval.id === action.id) {
          return {
            ...interval,
            ...action.updates
          }
        } else {
          return interval
        }
      })
    case 'DELETE_INTERVAL':
      return state.filter((interval) => {
        return interval.id !== action.id
      })
    case 'SET_INTERVALS':
      return action.intervals
    default:
      return state
  }
}