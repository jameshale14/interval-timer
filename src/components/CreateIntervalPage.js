import React from 'react'
import { connect } from 'react-redux'
import IntervalForm from './IntervalForm'

export class CreateIntervalPage extends React.Component {
  onSubmit = (interval) => {
    this.props.startCreateInterval(interval)
    this.props.history.push('/')
  }

  render() {
    return (
      <div>
        <IntervalForm onSubmit={this.onSubmit} />
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  startCreateInterval: (interval) => dispatch(startCreateInterval(interval))
})

export default connect(undefined, mapDispatchToProps)(CreateIntervalPage)