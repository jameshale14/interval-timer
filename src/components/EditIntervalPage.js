import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import IntervalForm from './IntervalForm'
import { startUpdateInterval } from '../actions/intervals'

export class EditIntervalPage extends React.Component {

  onSubmit = (interval) => {
    this.props.startUpdateInterval(this.props.interval.id, interval)
    this.props.history.push('/')
  }

  render() {
    return (
      <>
        <h3>Edit Interval</h3>
        <div>
          <IntervalForm onSubmit={this.onSubmit} interval={this.props.interval} />
        </div>
      </>
    )
  }
}

EditIntervalPage.propTypes = {
  interval: PropTypes.object,
  startUpdateInterval: PropTypes.func,
  history: PropTypes.object
}

const mapStateToProps = (state, props) => ({
  interval: state.intervals.find((interval) => interval.id === props.match.params.id)
})

const mapDispatchToProps = (dispatch) => ({
  startUpdateInterval: (id, interval) => dispatch(startUpdateInterval(id, interval))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditIntervalPage)