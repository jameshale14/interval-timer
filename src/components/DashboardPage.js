import React from 'react'
import { Link } from 'react-router-dom'
import IntervalList from './IntervalList'

const DashBoardPage = () => (
  <div>
    <Link className='button' to='/create'>New Interval</Link>
    <IntervalList />
  </div>
)

export default DashBoardPage 