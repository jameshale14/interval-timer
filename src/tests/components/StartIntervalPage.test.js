import React from 'react'
import { shallow } from 'enzyme'
import { StartIntervalPage } from '../../components/StartIntervalPage'
import { intervals } from '../fixtures/intervals'
import {
  oscillatorConnect,
  oscillatorStart,
  createOscillator,
  gainConnect,
  createGain,
  AudioContext,
  exponentialRampToValueAtTime
} from '../__mocks__/AudioContext'

let wrapper

window.AudioContext = AudioContext

beforeEach(() => {
  // Use these two for mocking the intervals
  jest.clearAllTimers()
  jest.useFakeTimers()

  window.AudioContext.mockClear()
  createOscillator.mockClear()
  createGain.mockClear()
  oscillatorConnect.mockClear()
  oscillatorStart.mockClear()
  gainConnect.mockClear()
  exponentialRampToValueAtTime.mockClear()

  wrapper = shallow(<StartIntervalPage interval={intervals[0]} />)
})

afterEach(() => {
  jest.clearAllTimers()
})

test('StartIntervalPage should render correctly', () => {
  expect(wrapper).toMatchSnapshot()
})


test('Interval should start when button is clicked', () => {
  wrapper.find('button').at(0).simulate('click', {
    preventDefault: () => { }
  })
  expect(setInterval).toHaveBeenCalledTimes(1)
  expect(setInterval).toHaveBeenCalledWith(expect.any(Function), 1000)
})


test('Interval should change state current step and time remaining when launched', () => {
  //start interval
  wrapper.find('button').at(0).simulate('click', {
    preventDefault: () => { }
  })

  expect(wrapper.state().currentStepIndex).toEqual(0)
  expect(wrapper.state().currentStepName).toEqual(intervals[0].steps[0].name)
  expect(wrapper.state().currentTimeRemaining).toEqual(intervals[0].steps[0].duration)
  expect(wrapper).toMatchSnapshot()

})


test('Interval should clear when stop button is clicked', () => {
  //start interval
  wrapper.find('button').at(0).simulate('click', {
    preventDefault: () => { }
  })
  //stop interval
  wrapper.find('button').at(1).simulate('click', {
    preventDefault: () => { }
  })

  expect(clearInterval).toHaveBeenCalledTimes(1)
})


test('Time remaining of interval step should reduce by 1 second for each second the interval runs for', () => {
  //start interval
  wrapper.find('button').at(0).simulate('click', {
    preventDefault: () => { }
  })

  jest.advanceTimersByTime(1000)

  expect(wrapper.state().currentStepIndex).toEqual(0)
  expect(wrapper.state().currentStepName).toEqual(intervals[0].steps[0].name)
  expect(wrapper.state().currentTimeRemaining).toEqual((intervals[0].steps[0].duration - 1))

  expect(wrapper).toMatchSnapshot()
})


test('When the timer reaches zero, the next second should set the interval step name and the time remaining to that of the next step', () => {
  //start interval
  wrapper.find('button').at(0).simulate('click', {
    preventDefault: () => { }
  })

  jest.advanceTimersByTime(1000 * (intervals[0].steps[0].duration + 1))

  expect(wrapper.state().currentStepIndex).toEqual(1)
  expect(wrapper.state().currentStepName).toEqual(intervals[0].steps[1].name)
  expect(wrapper.state().currentTimeRemaining).toEqual((intervals[0].steps[1].duration))
  expect(wrapper).toMatchSnapshot()
})


test('Should stop the interval after all of the steps have completed', () => {
  //start interval
  wrapper.find('button').at(0).simulate('click', {
    preventDefault: () => { }
  })

  for (let i = 0; i < intervals[0].steps.length; i++) {
    jest.advanceTimersByTime(1000 * (intervals[0].steps[i].duration + 1))
  }

  const lastStepIndex = intervals[0].steps.length - 1
  expect(wrapper.state().currentStepIndex).toEqual(lastStepIndex)
  expect(wrapper.state().currentStepName).toEqual(intervals[0].steps[lastStepIndex].name)
  expect(wrapper.state().currentTimeRemaining).toEqual(0)
  expect(setInterval).toHaveBeenCalledTimes(1)
  expect(clearInterval).toHaveBeenCalledTimes(1)
  expect(wrapper).toMatchSnapshot()

})

test('Stopping the timer should stop the time remaining from decreasing', () => {
  //start interval
  wrapper.find('button').at(0).simulate('click', {
    preventDefault: () => { }
  })

  jest.advanceTimersByTime(2000)

  //stop interval
  wrapper.find('button').at(1).simulate('click', {
    preventDefault: () => { }
  })

  jest.advanceTimersByTime(4000)

  expect(wrapper.state().currentStepIndex).toEqual(0)
  expect(wrapper.state().currentStepName).toEqual(intervals[0].steps[0].name)
  expect(wrapper.state().currentTimeRemaining).toEqual((intervals[0].steps[0].duration - 2))

  expect(wrapper).toMatchSnapshot()
})


test('interval should resume from current time remaining when start button is clicked after being stopped', () => {
  //start interval
  wrapper.find('button').at(0).simulate('click', {
    preventDefault: () => { }
  })

  jest.advanceTimersByTime(2000)

  //stop interval
  wrapper.find('button').at(1).simulate('click', {
    preventDefault: () => { }
  })

  //resume interval
  wrapper.find('button').at(0).simulate('click', {
    preventDefault: () => { }
  })

  jest.advanceTimersByTime(2000)

  expect(wrapper.state().currentStepIndex).toEqual(0)
  expect(wrapper.state().currentStepName).toEqual(intervals[0].steps[0].name)
  expect(wrapper.state().currentTimeRemaining).toEqual((intervals[0].steps[0].duration - 4))

  expect(wrapper).toMatchSnapshot()

})


test('"Restart Step" button should reinitialise current step', () => {
  //start interval
  wrapper.find('button').at(0).simulate('click', {
    preventDefault: () => { }
  })

  jest.advanceTimersByTime(25000)

  //stop interval
  wrapper.find('button').at(1).simulate('click', {
    preventDefault: () => { }
  })

  //re-start interval step
  wrapper.find('button').at(2).simulate('click', {
    preventDefault: () => { }
  })

  expect(wrapper.state().currentStepIndex).toEqual(1)
  expect(wrapper.state().currentStepName).toEqual(intervals[0].steps[1].name)
  expect(wrapper.state().currentTimeRemaining).toEqual((intervals[0].steps[1].duration))
  expect(wrapper).toMatchSnapshot()
})


test('"Restart Step" button should reinitialise current step', () => {
  //start interval
  wrapper.find('button').at(0).simulate('click', {
    preventDefault: () => { }
  })

  jest.advanceTimersByTime(25000)

  //stop interval
  wrapper.find('button').at(1).simulate('click', {
    preventDefault: () => { }
  })

  //re-start interval step
  wrapper.find('button').at(3).simulate('click', {
    preventDefault: () => { }
  })

  expect(wrapper.state().currentStepIndex).toEqual(0)
  expect(wrapper.state().currentStepName).toEqual(intervals[0].steps[0].name)
  expect(wrapper.state().currentTimeRemaining).toEqual((intervals[0].steps[0].duration))
  expect(wrapper).toMatchSnapshot()
})

test('when the timer reaches 3, 2 and 1 second remaining, there should be an audio signal created', () => {
  //start interval
  wrapper.find('button').at(0).simulate('click', {
    preventDefault: () => { }
  })

  jest.advanceTimersByTime(16000)

  expect(wrapper.state().currentTimeRemaining).toEqual(4)
  expect(gainConnect).toHaveBeenCalledTimes(0)
  expect(oscillatorConnect).toHaveBeenCalledTimes(0)
  expect(oscillatorStart).toHaveBeenCalledTimes(0)
  expect(exponentialRampToValueAtTime).toHaveBeenCalledTimes(0)

  jest.advanceTimersByTime(1000)

  expect(wrapper.state().currentTimeRemaining).toEqual(3)
  expect(gainConnect).toHaveBeenCalledTimes(1)
  expect(oscillatorConnect).toHaveBeenCalledTimes(1)
  expect(oscillatorStart).toHaveBeenCalledTimes(1)
  expect(exponentialRampToValueAtTime).toHaveBeenCalledTimes(1)
  expect(exponentialRampToValueAtTime).toHaveBeenLastCalledWith(0.00001, 1)

  jest.advanceTimersByTime(1000)
  expect(wrapper.state().currentTimeRemaining).toEqual(2)
  expect(gainConnect).toHaveBeenCalledTimes(2)
  expect(oscillatorConnect).toHaveBeenCalledTimes(2)
  expect(oscillatorStart).toHaveBeenCalledTimes(2)
  expect(exponentialRampToValueAtTime).toHaveBeenCalledTimes(2)
  expect(exponentialRampToValueAtTime).toHaveBeenLastCalledWith(0.00001, 1)

  jest.advanceTimersByTime(1000)
  expect(wrapper.state().currentTimeRemaining).toEqual(1)
  expect(gainConnect).toHaveBeenCalledTimes(3)
  expect(oscillatorConnect).toHaveBeenCalledTimes(3)
  expect(oscillatorStart).toHaveBeenCalledTimes(3)
  expect(exponentialRampToValueAtTime).toHaveBeenCalledTimes(3)
  expect(exponentialRampToValueAtTime).toHaveBeenLastCalledWith(0.00001, 1)
})

test('when the timer reaches zero the sound should last for longer', () => {

  //start interval
  wrapper.find('button').at(0).simulate('click', {
    preventDefault: () => { }
  })

  jest.advanceTimersByTime(20000)
  expect(wrapper.state().currentTimeRemaining).toEqual(0)
  expect(gainConnect).toHaveBeenCalledTimes(4)
  expect(oscillatorConnect).toHaveBeenCalledTimes(4)
  expect(oscillatorStart).toHaveBeenCalledTimes(4)
  expect(exponentialRampToValueAtTime).toHaveBeenCalledTimes(4)
  expect(exponentialRampToValueAtTime).toHaveBeenLastCalledWith(0.00001, 2.5)
})