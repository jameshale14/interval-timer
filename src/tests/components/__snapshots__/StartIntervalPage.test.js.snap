// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"Restart Interval" button should reinitialise whole routine 1`] = `
<div>
  <div>
    <h3
      data-testid="interval-name"
    >
      fitness workout 1
    </h3>
    <h1
      data-testid="step-name"
    >
      press-ups
    </h1>
    <h1
      data-testid="time-remaining"
    >
      20
    </h1>
    <button>
      Start
    </button>
    <button>
      Stop
    </button>
  </div>
</div>
`;

exports[`"Restart Step" button should reinitialise current step 1`] = `
<div>
  <div>
    <h3
      data-testid="interval-name"
    >
      fitness workout 1
    </h3>
    <h1
      data-testid="step-name"
    >
      Rest
    </h1>
    <h1
      data-testid="time-remaining"
    >
      10
    </h1>
    <button>
      Start
    </button>
    <button>
      Stop
    </button>
  </div>
</div>
`;

exports[`Interval should change state current step and time remaining when launched 1`] = `
<div>
  <div>
    <h3
      data-testid="interval-name"
    >
      fitness workout 1
    </h3>
    <h1
      data-testid="step-name"
    >
      press-ups
    </h1>
    <h1
      data-testid="time-remaining"
    >
      20
    </h1>
    <button>
      Start
    </button>
    <button>
      Stop
    </button>
  </div>
</div>
`;

exports[`Should stop the interval after all of the steps have completed 1`] = `
<div>
  <div>
    <h3
      data-testid="interval-name"
    >
      fitness workout 1
    </h3>
    <h1
      data-testid="step-name"
    >
      Rest
    </h1>
    <h1
      data-testid="time-remaining"
    >
      0
    </h1>
    <button>
      Start
    </button>
    <button>
      Stop
    </button>
    <button>
      Restart Step
    </button>
    <button>
      Restart Interval
    </button>
  </div>
</div>
`;

exports[`StartIntervalPage should render correctly 1`] = `
<div>
  <div>
    <h3
      data-testid="interval-name"
    >
      fitness workout 1
    </h3>
    <h1
      data-testid="step-name"
    />
    <h1
      data-testid="time-remaining"
    />
    <button>
      Start
    </button>
    <button>
      Stop
    </button>
  </div>
</div>
`;

exports[`Stopping the timer should stop the time remaining from decreasing 1`] = `
<div>
  <div>
    <h3
      data-testid="interval-name"
    >
      fitness workout 1
    </h3>
    <h1
      data-testid="step-name"
    >
      press-ups
    </h1>
    <h1
      data-testid="time-remaining"
    >
      18
    </h1>
    <button>
      Start
    </button>
    <button>
      Stop
    </button>
    <button>
      Restart Step
    </button>
    <button>
      Restart Interval
    </button>
  </div>
</div>
`;

exports[`Time remaining of interval step should reduce by 1 second for each second the interval runs for 1`] = `
<div>
  <div>
    <h3
      data-testid="interval-name"
    >
      fitness workout 1
    </h3>
    <h1
      data-testid="step-name"
    >
      press-ups
    </h1>
    <h1
      data-testid="time-remaining"
    >
      19
    </h1>
    <button>
      Start
    </button>
    <button>
      Stop
    </button>
  </div>
</div>
`;

exports[`When the timer reaches zero, the next second should set the interval step name and the time remaining to that of the next step 1`] = `
<div>
  <div>
    <h3
      data-testid="interval-name"
    >
      fitness workout 1
    </h3>
    <h1
      data-testid="step-name"
    >
      Rest
    </h1>
    <h1
      data-testid="time-remaining"
    >
      10
    </h1>
    <button>
      Start
    </button>
    <button>
      Stop
    </button>
  </div>
</div>
`;

exports[`interval should resume from current time remaining when start button is clicked after being stopped 1`] = `
<div>
  <div>
    <h3
      data-testid="interval-name"
    >
      fitness workout 1
    </h3>
    <h1
      data-testid="step-name"
    >
      press-ups
    </h1>
    <h1
      data-testid="time-remaining"
    >
      16
    </h1>
    <button>
      Start
    </button>
    <button>
      Stop
    </button>
  </div>
</div>
`;
